web/src/app/components/tasks-editor/task-condition-form/task-condition-form.component.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2024 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, Input, Pipe, PipeTransform} from '@angular/core';
+import {AbstractControl, FormArray, FormGroup} from '@angular/forms';
+import {List} from 'immutable';
+
+import {Option} from 'app/models/task/option.model';
+import {Task} from 'app/models/task/task.model';
+
+@Pipe({name: 'getTaskOptions', pure: false})
+export class TaskOptionsPipe implements PipeTransform {
+  transform(tasks: List<Task>, filter: string): List<Option> {
+    return (
+      tasks.find(task => task.id === filter)?.multipleChoice?.options ||
+      List([])
+    );
+  }
+}
+
+@Component({
+  selector: 'ground-task-condition-form',
+  templateUrl: './task-condition-form.component.html',
+  styleUrls: ['./task-condition-form.component.scss'],
+})
+export class TaskConditionFormComponent {
+  @Input() formGroup!: FormGroup;
+  @Input() formGroupIndex!: number;
+  @Input() tasks!: List<Task>;
+
+  previousMultipleTasks: List<Task> = List([]);
+  options: List<Task> = List([]);
+
+  ngOnChanges() {
+    this.previousMultipleTasks = this.tasks.filter(
```

Comment:
There are two types of tasks in this process:

- Prerequisite Tasks: These tasks must be completed before a specific task can appear.ù
- Conditional Tasks: These tasks are only presented if a certain condition is met.

The order of the tasks is crucial because the app presents them sequentially. This means completing the prerequisite tasks in the correct order is necessary for the conditional task to appear.